src/server/zset_family.cc

```diff
@@ -2445,6 +2468,39 @@ void ZSetFamily::GeoHash(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void ZSetFamily::GeoPos(CmdArgList args, ConnectionContext* cntx) {
+  string_view key = ArgS(args, 0);
+
+  absl::InlinedVector<string_view, 8> members(args.size() - 1);
+  for (size_t i = 1; i < args.size(); ++i) {
+    members[i - 1] = ArgS(args, i);
+  }
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    return OpMScore(t->GetOpArgs(shard), key, members);
+  };
+
+  OpResult<MScoreResponse> result = cntx->transaction->ScheduleSingleHopT(std::move(cb));
+
+  if (result.status() == OpStatus::WRONG_TYPE) {
+    return (*cntx)->SendError(kWrongTypeErr);
+  }
+
+  (*cntx)->StartArray(result->size());  // Array return type.
+  const MScoreResponse& arr = result.value();
```

Comment:
`result` could have a non-ok status for other reasons (i.e. not `WRONG_TYPE`). Let's make sure we handle these as well, as otherwise trying to access `.value()` can lead to a crash or worse.

Reply:
Could you provide some guidance as to what errors I should address?

Reply:
I don't think you should _special_ handle any other error states. But I do think you should add some general if-error-return-error handler to avoid invalid memory access / undefined behavior.

Reply:
I saw that in case of an error, an error string is returned in the following way:
```if (result.status() != OpStatus::OK) {
    return (*cntx)->SendError(kWrongTypeErr);
  }
  ```
So I tried creating a generic function that maps the error in the `OpStatus` enum to the error strings defined in `facade`, but It seems like some errors don't have any error strings, or require a different behavior. For example, seems like the `OpStatus::SKIPPED` error usually requires different handling than error strings - returning null, skipping, etc.
I was thinking about implementing a `handleOpStatusError` function, but this may be out of the scope of this issue. WDYT? @chakaz

Reply:
We do have `SendError` that receives OpStatus as an argument. We, indeed, use it for the subset of common statuses. All the command-specific errors are handled explicitly.

Reply:
Ok great!
@romange @chakaz Since I'm not familiar with the `GeoPos` command, could you provide some guidance as to what command-specific errors should I handle? My current implementation is based on what I saw in Redis's implementation - only verifying the type of the key.

Reply:
Any reason not to simply add
```cpp
} else if (!result) {
  return (*cntx)->SendError(result.status());
}
```
Like @romange suggested?

Anyway, we can add specific cases later in case we missed something, what's most important is to not crash on errors.

Reply:
@chakaz added your suggestion :) LMK if there are any other comments